lib/components/tabs/tabs.dart

```diff
@@ -0,0 +1,76 @@
+import 'dart:developer';
+import 'dart:html';
+import 'package:katikati_ui_lib/components/logger.dart';
+
+var logger = Logger('Tabs');
+
+class TabsView {
+  DivElement renderElement;
+
+  List<String> _tabLabels;
+  List<DivElement> _tabContents;
+  int _selectedIndex;
+  String _tabContentClassname;
+
+  List<SpanElement> _tabChoosers = [];
+  DivElement _tabContent;
+
+  TabsView(this._tabLabels, this._tabContents, {int defaultSelectedIndex = 0, String tabContentClassname}) {
```

Comment:
I'm a bit wary of separating labels and contents into different lists, and then matching them up by index. You end up needing to do the verification that you're doing below, and you won't know about it until you run the code, rather than when you're editing it.

What I'd suggest we do in this case is to have an abstract `TabView` class that defines the minimum things that a tab will need in order for the `TabsView` class to be able to use it, and then pass in here the list to it? The tabs can then implement this abstract class and extend it as they need.

Here's a rough sketch of how it might look like:
```
class TabsView {
  DivElement renderElement;
  List<TabView> tabs;

  TabsView(this.tabs, ...) {...}
}

abstract class TabView {
  String id; // to be used by the TabsView to identify the tab (usually more stable than an index)
  String label; // to be used by the TabsView to populate the list of tabs
  DivElement renderElement; // the contents of the tab, to be instantiated and populated by the implementing class
  
  TabView(this.id, this.label);
}
```

Reply:
Yep this makes sense to use id as the primary selector instead of the index. Updated the code to use id, but I'm using find to get the index of the tab and update the dom. Please let me know if I shouldn't do this and find dom elements by id and update them.